### ISSUE TITLE
[tune] Support for passing strings into config during PBT

### ISSUE THREAD

**Issue by arsedler9** [ray-project/ray#6792]
I think it would be useful to be able to pass strings into `tune.Trainable` subclasses through the config dictionary (e.g. directories). This works for random search, but seems to be broken for PBT because PBT tries to mutate everything in the config dict.

**Comment by hhbyyh** [comment]
In my experiment, `pbt` can take a string list in config and hyperparam_mutations, e.g. 
```
hyperparam_mutations={
            "lr": lambda: np.random.uniform(0.0001, 1),
            "momentum": [0.8, 0.9, 0.99],
            "something": ["a", "b"]
        }
```

`pbt` can mutate the string param by choosing adjacent item in list with specified probability. Not sure if this is what you want. Could you please post some code examples?

**Comment by arsedler9** [comment]
Thanks for your help @hhbyyh. I was actually referring to the config passed to `tune.run` rather than the hp mutations. I've since modified my own code for a workaround, but I tested by adding an additional key to the config parameter of `tune.run` in `pbt_example.py` and it seems to work fine. Will reopen the issue if I run into this again. Thanks! 
```
config={
    "lr": 0.0001,
    "some_other_factor": 1,
    "yaml": 'default.yaml',
```
